src/goto-programs/goto_contractor.cpp

```diff
@@ -0,0 +1,453 @@
+  if(
+    is_arith_expr(expr) || is_constant_number(expr) || is_symbol2t(expr) ||
+    is_notequal2t(expr))
+  {
+    message_handler.status("Expression is complex, skipping this assert");
+  }
+
+  std::shared_ptr<relation_data> rel;
+  rel = std::dynamic_pointer_cast<relation_data>(expr);
+
+  Function *f, *g;
+  f = create_function_from_expr2t(rel->side_1);
+  g = create_function_from_expr2t(rel->side_2);
+  switch(expr->expr_id)
+  {
+  case expr2t::expr_ids::greaterthanequal_id:
+    c = new NumConstraint(*vars, (*f)(*vars) >= (*g)(*vars));
+    break;
+  case expr2t::expr_ids::greaterthan_id:
+    c = new NumConstraint(*vars, (*f)(*vars) > (*g)(*vars));
+    break;
+  case expr2t::expr_ids::lessthanequal_id:
+    c = new NumConstraint(*vars, (*f)(*vars) <= (*g)(*vars));
+    break;
+  case expr2t::expr_ids::lessthan_id:
+    c = new NumConstraint(*vars, (*f)(*vars) < (*g)(*vars));
+    break;
+  case expr2t::expr_ids::equality_id:
+    c = new NumConstraint(*vars, (*f)(*vars) = (*g)(*vars));
+    break;
+  default:
+    break;
+  }
+  ///keeping this for later
+  /*
+  if(is_greaterthanequal2t(expr))
+  {
+    Function *f, *g;
+    f = create_function_from_expr2t(to_greaterthanequal2t(expr).side_1);
+    g = create_function_from_expr2t(to_greaterthanequal2t(expr).side_2);
+    c = new NumConstraint(*vars, (*f)(*vars) >= (*g)(*vars));
+    return c;
+  }
+  else if(is_greaterthan2t(expr))
+  {
+    Function *f, *g;
+    f = create_function_from_expr2t(to_greaterthan2t(expr).side_1);
+    g = create_function_from_expr2t(to_greaterthan2t(expr).side_2);
+    c = new NumConstraint(*vars, (*f)(*vars) > (*g)(*vars));
+    return c;
+  }
+  else if(is_lessthanequal2t(expr))
+  {
+    Function *f, *g;
+    f = create_function_from_expr2t(to_lessthanequal2t(expr).side_1);
+    g = create_function_from_expr2t(to_lessthanequal2t(expr).side_2);
+    c = new NumConstraint(*vars, (*f)(*vars) <= (*g)(*vars));
+    return c;
+  }
+  else if(is_lessthan2t(expr))
+  {
+    Function *f, *g;
+    f = create_function_from_expr2t(to_lessthan2t(expr).side_1);
+    g = create_function_from_expr2t(to_lessthan2t(expr).side_2);
+    c = new NumConstraint(*vars, (*f)(*vars) < (*g)(*vars));
+    return c;
+  }
+  else if(is_equality2t(expr))
+  {
+    Function *f, *g;
+    f = create_function_from_expr2t(to_equality2t(expr).side_1);
+    g = create_function_from_expr2t(to_equality2t(expr).side_2);
+    c = new NumConstraint(*vars, (*f)(*vars) = (*g)(*vars));
+    return c;
+  }
+  else if(is_notequal2t(expr))
+  {
+    Function *f, *g;
+    f = create_function_from_expr2t(to_notequal2t(expr).side_1);
+    g = create_function_from_expr2t(to_notequal2t(expr).side_2);
+    c = new NumConstraint(*vars, (*f)(*vars) = (*g)(*vars));
+    return c;
+  }*/
+
+  return c;
+}
+
+Function *
+goto_contractort::create_function_from_expr2t(irep_container<expr2t> expr)
+{
+  Function *f = nullptr;
+  Function *g, *h;
+
+  if(is_comp_expr(expr))
+  {
+    //Abort contractor
```

Comment:
?

Reply:
Updated to return nullptr so we can resume verification if goto-contractor is not applicable.